app/services/team_users/build_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module TeamUsers
+  class BuildService < ::BaseService
+    def execute
+      team_user = build_team_user unless params.empty?
+
+      enforce_admin_owner!(team_user) unless team_user.nil?
+
+      team_user
+    end
+
+    private
+
+    # Always set role as owner if user is a Portus admin
+    def enforce_admin_owner!(team_user)
+      team_user[:role] = TeamUser.roles[:owner] if team_user.user&.admin?
+    end
+
+    def build_team_user
+      team = fetch_team
+      user = fetch_user
+      TeamUser.new(user: user, team: team, role: params[:role])
+    end
+
+    def fetch_user
+      user = User.find_by(username: params[:user])
+      params.delete(:user)
+      raise ActiveRecord::RecordNotFound if user.nil?
```

Comment:
You don't have to do this if you call `find_by!` instead of `find_by` before :wink:

Reply:
Done. I've also fixed this in other places regarding this PR.